### PR TITLE
feat(mentee-page): add mentee-page, lesson components

### PR DESCRIPTION
**Pull request by iradamamedova**
made component mentee-page, which includes other components lessons

### CODE REVIEW

**Comment by Tsurhanau** [comment]
Please name Pull request according to the rules:
https://github.com/JS-Machine-University/JSMU-WEB/wiki/Git-rules#merge-requests-params

File: projects/common-components/karma.conf.js [hunk 1]
```diff
@@ -19,26 +19,23 @@ module.exports = function (config) {
...
     colors: true,
     logLevel: config.LOG_INFO,
     autoWatch: true,
-    browsers: ['Chrome'],
+    browsers: ["Chrome"],
     singleRun: false,
-    restartOnFileChange: true
+    restartOnFileChange: true,
   });
 };
```
Comment (Tsurhanau) [inline projects/common-components/karma.conf.js:41]: changes in this files do not apply to your task, perhaps the reason is the difference in the prettier settings, please check this

changes that are not related to the task should not be commit to the PR

File: projects/core/src/lib/mentee-page/mentee-page.component.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<div class="lessons">
+	<h2 class="lessons__title">Modules</h2>
+	<ng-container *ngFor="let lesson of lessons">
+		<jsmu-lesson [lesson]="lesson"></jsmu-lesson>
+		<hr />
```
Comment (Tsurhanau) [inline projects/core/src/lib/mentee-page/mentee-page.component.html:5]: hide last item

File: projects/core/src/lib/services/lessons.data.service.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from "@angular/core";
+import { Observable } from "rxjs";
+import { DataBaseService } from "./database.service";
+import { Lesson } from "../models/lesson";
+import { ListType } from "../models/list-type";
+import { HttpClient } from "@angular/common/http";
+
+@Injectable()
+export class LessonsDataService extends DataBaseService<Lesson> {
+	public getLesson<Lesson>(): Observable<Lesson> {
```
Comment (Tsurhanau) [inline projects/core/src/lib/services/lessons.data.service.ts:10]: please add tests

File: src/app/app.module.ts [hunk 4]
```diff
@@ -9,14 +9,17 @@ import { environment } from "src/environments/environment";
...
 import { StoreModule } from "@ngrx/store";
 import { StoreDevtoolsModule } from "@ngrx/store-devtools";
+import { MenteePageComponent } from "projects/core/src/lib/mentee-page/mentee-page.component";
+import { LessonComponent } from "projects/core/src/lib/lesson/lesson.component";
+import { LessonsDataService } from "projects/core/src/lib/services/lessons.data.service";
+import { DataBaseService } from "projects/core/src/lib/services/database.service";
 
 @NgModule({
-	declarations: [AppComponent],
+	declarations: [AppComponent, MenteePageComponent, LessonComponent],
```
[the remark this answers is not in the document]
Comment (iradamamedova) [reply src/app/app.module.ts:22]: Done

File: projects/core/src/lib/mentee-page/mentee-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	selector: "jsmu-mentee-page",
+	templateUrl: "./mentee-page.component.html",
+	styleUrls: ["./mentee-page.component.scss"]
+})
+export class MenteePageComponent implements OnInit {
+	public lessons: Lesson[] = [];
+
+	constructor(private ldb: LessonsDataService) {}
+
+	private saveLesson(lesson: Lesson): Observable<Lesson> {
```
[the remark this answers is not in the document]
Comment (iradamamedova) [reply projects/core/src/lib/mentee-page/mentee-page.component.ts:16]: Done

File: projects/core/src/lib/mentee-page/mentee-page.component.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+		return this.ldb.saveLesson(lesson);
+	}
+	private getLessons(): void {
+		this.ldb.getLesson().subscribe((lessons: any) => {
```
[the remark this answers is not in the document]
Comment (iradamamedova) [reply projects/core/src/lib/mentee-page/mentee-page.component.ts:20]: done

File: projects/core/src/lib/mentee-page/mentee-page.component.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+			lessons.map((lesson: Lesson) => {
+				if (lesson !== null) {
```
[the remark this answers is not in the document]
Comment (iradamamedova) [reply projects/core/src/lib/mentee-page/mentee-page.component.ts:22]: resolved

File: projects/core/src/lib/mentee-page/mentee-page.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+		component = fixture.componentInstance;
+		fixture.detectChanges();
+	});
+
+	it("should create", () => {
+		expect(component).toBeDefined();
+	});
+
+	it("should be able to save Lessons", () => {
+		expect(component["saveLesson"](lesson1)).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (iradamamedova) [reply projects/core/src/lib/mentee-page/mentee-page.component.spec.ts:33]: i made it public, cause it'll probably be used in another place

File: projects/core/src/lib/mentee-page/mentee-page.component.scss [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+	font-family: "Source Sans Pro", sans-serif;
+	background: #fff;
+}
+.lessons {
+	&__title {
+		color: #444;
+		letter-spacing: 0.03em;
+		margin: 20px 0;
+	}
+	hr {
```
[the remark this answers is not in the document]
Comment (iradamamedova) [reply projects/core/src/lib/mentee-page/mentee-page.component.scss:14]: resolved

File: projects/core/src/lib/lesson/lesson.component.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+@Component({
+	selector: "jsmu-lesson",
+	templateUrl: "./lesson.component.html",
+	styleUrls: ["./lesson.component.scss"]
+})
+export class LessonComponent {
+	public selected: boolean = false;
+
+	@Input()
+	public lesson: Lesson | undefined;
```
[the remark this answers is not in the document]
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.ts:13]: done

File: projects/core/src/lib/lesson/lesson.component.ts [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+	public select(lesson: Lesson): void {
```
[the remark this answers is not in the document]
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.ts:15]: resolved

File: projects/core/src/lib/mentee-page/mentee-page.component.html (hunk 2 above)
Comment (iradamamedova) [reply projects/core/src/lib/mentee-page/mentee-page.component.html:5]: done

File: projects/common-components/karma.conf.js (hunk 1 above)
Comment (iradamamedova) [reply projects/common-components/karma.conf.js:41]: ok

File: projects/core/src/lib/lesson/lesson.component.scss [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+		}
+	}
+	&__details {
+		margin: 20px;
+	}
+	&__materials {
+		&__title,
+		&__description,
+		&__link {
+			margin: 10px 0;
```
[the remark this answers is not in the document]
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.scss:38]: done

File: projects/core/src/lib/lesson/lesson.component.html [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+		<h4 class="lesson__title">
+			<i class="fa-solid fa-chevron-down" *ngIf="!selected"></i>
+			<i class="fa-solid fa-chevron-up" *ngIf="selected"></i>
+			Module #{{ lesson.id }}: {{ lesson.title }}
+		</h4>
+		<div class="lesson__details" *ngIf="selected">
+			<hr />
+			<div class="lesson__materials">
+				<h4 class="lesson__title">Self-study Materials</h4>
+				<p class="lesson__materials__description">
```
[the remark this answers is not in the document]
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.html:12]: done

File: projects/core/src/lib/services/lessons.data.service.ts (hunk 3 above)
Comment (iradamamedova) [reply projects/core/src/lib/services/lessons.data.service.ts:10]: done

File: projects/core/src/lib/lesson/lesson.component.scss [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	&__separator {
+		border: none;
+		border-top: 1px solid #fafafa;
+		margin: 0;
+	}
+	&__title {
+		@extend %vertical-margin;
+	}
+	.resources {
```
Comment (Tsurhanau) [inline projects/core/src/lib/lesson/lesson.component.scss:42]: please move this block inside of lesson block

File: projects/core/src/lib/lesson/lesson.component.scss [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+		&__description {
+			@extend %vertical-margin;
+		}
+		.resource {
```
Comment (Tsurhanau) [inline projects/core/src/lib/lesson/lesson.component.scss:46]: please make this class part of BEM structure

File: projects/core/src/lib/lesson/lesson.component.html [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+				</p>
+				<p class="resources__resource resource" *ngFor="let resource of lesson.resources">
+					<i class="resource__paperclip fa-solid fa-paperclip" *ngIf="resource"></i>
+					<a class="resource__link" href="{{ resource.url }}" target="_blank">
+						{{ resource.name }}
+					</a>
+				</p>
+			</div>
+			<hr class="lesson__separator" />
+			<div class="lesson__meeting meeting">
```
Comment (Tsurhanau) [inline projects/core/src/lib/lesson/lesson.component.html:24]: .meeting class is missing from component's style rules

File: projects/core/src/lib/lesson/lesson.component.scss [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+			&__link {
+				text-decoration: none;
+				color: #009ecc;
+				&:hover {
+					color: #008abd;
+				}
+			}
+		}
+	}
+	.duration {
```
Comment (Tsurhanau) [inline projects/core/src/lib/lesson/lesson.component.scss:56]: please make this class part of BEM structure

File: projects/core/src/lib/lesson/lesson.component.scss [hunk 18]
```diff
@@ -0,0 +1,66 @@
+@import url("https://fonts.googleapis.com/css2?family=Source+Sans+Pro:wght@400&&display=swap");
```
Comment (Tsurhanau) [inline projects/core/src/lib/lesson/lesson.component.scss:1]: maybe it is better way to put this import in styles.scss, so that the font is available to every component

File: projects/core/src/lib/mentee-page/mentee-page.component.scss [hunk 19]
```diff
@@ -0,0 +1,22 @@
+@import url("https://fonts.googleapis.com/css2?family=Source+Sans+Pro:wght@400&&display=swap");
+:host {
```
Comment (Tsurhanau) [inline projects/core/src/lib/mentee-page/mentee-page.component.scss:2]: instead of host you can add main block .mentee-page and apply these styles

File: projects/core/src/lib/mentee-page/mentee-page.component.spec.ts [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+
+	it("should create", () => {
+		expect(component).toBeTruthy();
+	});
+
+	it("should get lessons ", () => {
+		spyOn(ldb, "getLesson").and.callThrough();
+		expect(
+			ldb.getLesson<Lesson[]>().subscribe((lessons: Lesson[]) => {
+				component.lessons = lessons;
```
Comment (Tsurhanau) [inline projects/core/src/lib/mentee-page/mentee-page.component.spec.ts:34]: check that lessons corresponds to the expected object, after ldb returnValue

File: projects/core/src/lib/mentee-page/mentee-page.component.ts [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+export class MenteePageComponent implements OnInit {
+	public lessons: Lesson[] = [];
+
+	constructor(private ldb: LessonsDataService) {}
+
+	public saveLesson<Lesson>(lesson: Lesson): Observable<Lesson> {
+		return this.ldb.saveLesson(lesson);
+	}
+	public getLessons(): void {
+		this.ldb.getLesson<Lesson[]>().subscribe((lessons: Lesson[]) => {
```
Comment (Tsurhanau) [inline projects/core/src/lib/mentee-page/mentee-page.component.ts:20]: please unsubscribe from observable

File: projects/core/src/lib/lesson/lesson.component.scss (hunk 18 above)
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.scss:1]: but if I add it to the general style.scss file then everyone's font will change, because they are different

Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.scss:1]: done

File: projects/core/src/lib/lesson/lesson.component.scss (hunk 14 above)
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.scss:42]: it is, what do you mean? like lesson__resources instead? but there is another element inside resources block, so in any case i will need to write resouces like i did

File: projects/core/src/lib/lesson/lesson.component.scss (hunk 15 above)
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.scss:46]: can you please explain what you mean

File: projects/core/src/lib/lesson/lesson.component.html (hunk 16 above)
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.html:24]: ok, i removed this class

File: projects/core/src/lib/lesson/lesson.component.scss (hunk 17 above)
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.scss:56]: still don't understand what should i do with it

File: projects/core/src/lib/mentee-page/mentee-page.component.scss (hunk 19 above)
Comment (iradamamedova) [reply projects/core/src/lib/mentee-page/mentee-page.component.scss:2]: done

File: projects/core/src/lib/mentee-page/mentee-page.component.spec.ts (hunk 20 above)
Comment (iradamamedova) [reply projects/core/src/lib/mentee-page/mentee-page.component.spec.ts:34]: done

File: projects/core/src/lib/mentee-page/mentee-page.component.ts (hunk 21 above)
Comment (iradamamedova) [reply projects/core/src/lib/mentee-page/mentee-page.component.ts:20]: done

File: projects/core/src/lib/lesson/lesson.component.scss (hunk 14 above)
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.scss:40]: done

File: projects/core/src/lib/lesson/lesson.component.scss (hunk 17 above)
Comment (iradamamedova) [reply projects/core/src/lib/lesson/lesson.component.scss:54]: done
